### ISSUE TITLE
[feature request]: `selector-nest-combinators` option `least`

### ISSUE THREAD

**Issue by JounQin** [kristerkari/stylelint-scss#390]
First case which should emit error:

```scss
.a {
  &.b {
    color: red;
  }
}
```

Second case which should be allowed:
```scss
.a {
  &.b {
    color: red;
  }

  .c {
    font-size: 16px;
  }
}
```

@kristerkari What do you think?

**Comment by kristerkari** [comment]
Hi @JounQin and sorry for the late reply. 

I'm not totally sure what's the use case for having such option, could you provide another example that would maybe be related to some issue that you are having with the rule? 

Also, why is the option called `least`? Is that a good name to describe it?

I'm totally ok with adding the option, but I just need to understand it properly first.

**Comment by JounQin** [comment]
@kristerkari We don't want to add redundant nested block in our team, so current option `always | never` is both too strict and unreasonable for us.

```scss
// input
.a {
  &.b {
    color: red;
  }
}

// expected output
.a.b {
  color: red;
}
```

> Also, why is the option called least? Is that a good name to describe it?

Maybe `as-needed` be better or not?

**Comment by JounQin** [comment]
@kristerkari Sorry to interrupt, any further discussion?

**Comment by ilyub** [comment]
Any movement on this issue?

Motivation:

Consider the following css code:
```css
.group .item thead th { ... }
.group .item tbody td { ... }
.group .item tfoot td { ... }
```

I want to avoid duplicating ".group .item" part, but I don't want to add unnecessary nesting. I.e. this is what I want:
```css
.group .item {
  thead th { ... }
  tbody td { ... }
  tfoot td { ... }
}
```

Obviously neither "never" nor "always" option for selector-nest-combinators let me do this.

So, I agee that "as-needed" option would be helpful that would:
1. Require nesting for duplicate prefix (e.g. ".group .item" in my example above).
2. Forbid nesting in all other cases.

**Comment by kristerkari** [comment]
@ilyub I'm totally fine with adding such option. Are you able to contribute a Pull Request, or do you know someone who would be able to do it?

**Comment by ilyub** [comment]
I never dealt with stylelint rule programming. So, I can only hope that somebody else gets interested who is more familiar with this ecosystem.
